relayer/txm/txm_test.go

```diff
@@ -200,3 +314,1009 @@ func TestTxm(t *testing.T) {
+		time.Sleep(100 * time.Millisecond)
+
+		totalFinishedAfter := 0
+		for _, account := range accounts {
+			store := txm.AccountStore.GetTxStore(account)
+			totalFinishedAfter += store.FinishedCount()
+		}
+		require.Equal(t, 0, totalFinishedAfter)
+	})
+
+	t.Run("No reaping when no expired transactions", func(t *testing.T) {
+		combinedClient := createDefaultMockClient(t)
+
+		noReapConfig := &trontxm.TronTxmConfig{
+			BroadcastChanSize: 100,
+			ConfirmPollSecs:   1,
+			RetentionPeriod:   5 * time.Second,
+			ReapInterval:      50 * time.Millisecond,
+		}
+
+		txm, _, _ := setupTxm(t, combinedClient, noReapConfig)
+		defer txm.Close()
+
+		store := txm.AccountStore.GetTxStore(genesisAddress.String())
+
+		recentTx := &trontxm.TronTx{
+			ID:          "recent_tx",
+			FromAddress: genesisAddress,
+			CreateTs:    time.Now(),
+		}
+
+		require.NoError(t, store.OnPending("recent_hash", time.Now().UnixMilli()+1000, recentTx))
+		require.NoError(t, store.OnConfirmed(recentTx.ID))
+		require.NoError(t, store.OnFinalized(recentTx.ID))
+
+		require.Equal(t, 1, store.FinishedCount())
+
+		time.Sleep(200 * time.Millisecond)
+
+		require.True(t, store.Has(recentTx.ID))
+		require.Equal(t, 1, store.FinishedCount())
+	})
+
+	t.Run("Reap performance with many transactions", func(t *testing.T) {
+		combinedClient := createDefaultMockClient(t)
+
+		perfConfig := &trontxm.TronTxmConfig{
+			BroadcastChanSize: 100,
+			ConfirmPollSecs:   1,
+			RetentionPeriod:   100 * time.Millisecond,
+			ReapInterval:      50 * time.Millisecond,
+		}
+
+		txm, _, _ := setupTxm(t, combinedClient, perfConfig)
+		defer txm.Close()
+
+		store := txm.AccountStore.GetTxStore(genesisAddress.String())
+
+		numTxs := 100
+		for i := 0; i < numTxs; i++ {
+			tx := &trontxm.TronTx{
+				ID:          fmt.Sprintf("perf_tx_%d", i),
+				FromAddress: genesisAddress,
+				CreateTs:    time.Now(),
+			}
+
+			require.NoError(t, store.OnPending(fmt.Sprintf("perf_hash_%d", i), time.Now().UnixMilli()+1000, tx))
+			require.NoError(t, store.OnConfirmed(tx.ID))
+			require.NoError(t, store.OnFinalized(tx.ID))
+		}
+
+		require.Equal(t, numTxs, store.FinishedCount())
+
+		startTime := time.Now()
+		time.Sleep(200 * time.Millisecond)
+		reapDuration := time.Since(startTime)
+
+		require.Equal(t, 0, store.FinishedCount())
+		require.Less(t, reapDuration, 1*time.Second)
+	})
+}
+
+func TestTxmStateTransitions(t *testing.T) {
+	t.Parallel()
+
+	t.Run("TxStore state transitions", func(t *testing.T) {
+		combinedClient := createDefaultMockClient(t)
+		txm, _, _ := setupTxm(t, combinedClient, nil)
+		defer txm.Close()
+
+		store := txm.AccountStore.GetTxStore(genesisAddress.String())
+
+		// OnPending
+		tx1 := &trontxm.TronTx{ID: "id1", FromAddress: genesisAddress}
+		hash1 := "hash1"
+		require.NoError(t, store.OnPending(hash1, 1000, tx1))
+		require.True(t, store.Has("id1"))
+		require.Equal(t, trontxm.Pending, tx1.State)
+		status, err := txm.GetTransactionStatus(context.Background(), tx1.ID)
```

Comment:
nit: think we can just use `t.Context()` here and anywhere else in tests
```suggestion
		status, err := txm.GetTransactionStatus(t.Context(), tx1.ID)
```